src/app/core/entity/entity-remove.service.spec.ts

```diff
@@ -31,80 +32,66 @@ describe("EntityRemoveService", () => {
       providers: [
         { provide: EntityMapperService, useValue: mockEntityMapper },
         { provide: MatSnackBar, useValue: snackBarSpy },
+        Router,
         {
           provide: ConfirmationDialogService,
           useValue: mockConfirmationDialog,
         },
       ],
     });
-    service = TestBed.inject(EntityRemoveService);
-  });
+    mockRouter = TestBed.inject(Router);
+    spyOn(mockRouter, "navigate");
 
-  it("should be created", () => {
-    expect(service).toBeTruthy();
+    service = TestBed.inject(EntityRemoveService);
   });
 
-  it("emits once and closes when the user has cancelled", (done) => {
+  it("should return false when user cancels confirmation", async () => {
     mockConfirmationDialog.getConfirmation.and.resolveTo(false);
-    service
-      .remove(new Entity())
-      .pipe(toArray())
-      .subscribe({
-        next: (next) => {
-          expect(next).toEqual([RemoveResult.CANCELLED]);
-        },
-        complete: () => {
-          expect(snackBarSpy.open).not.toHaveBeenCalled();
-          expect(mockEntityMapper.remove).not.toHaveBeenCalled();
-          done();
-        },
-      });
+
+    const result = await service.remove(new Entity());
+
+    expect(result).toBe(false);
+    expect(snackBarSpy.open).not.toHaveBeenCalled();
+    expect(mockEntityMapper.remove).not.toHaveBeenCalled();
   });
 
-  it("deletes the entity and finishes if the action is never undone", (done) => {
+  it("should delete entity, show snackbar confirmation and navigate back", async () => {
     // onAction is never called
     mockSnackBarRef.onAction.and.returnValues(NEVER);
     // mock that dialog is dismissed immediately
```

Comment:
I am not sure what dismissing the snackbar means? In the UI there is no option to close the snackbar without clicking undo.